mlflow/pyfunc/__init__.py

```diff
@@ -485,30 +484,27 @@ def load_model(
     return PyFuncModel(model_meta=model_meta, model_impl=model_impl, predict_fn=predict_fn)
 
 
-def _download_model_conda_env(model_uri):
-    conda_yml_file_name = _get_flavor_configuration_from_uri(model_uri, FLAVOR_NAME)[ENV]
-    return _download_artifact_from_uri(append_to_uri_path(model_uri, conda_yml_file_name))
-
-
 def _get_model_dependencies(model_uri, format="pip"):  # pylint: disable=redefined-builtin
+    model_dir = _download_artifact_from_uri(model_uri)
+    conda_yaml_filename = _get_flavor_configuration_from_ml_model_file(
+        os.path.join(model_dir, MLMODEL_FILE_NAME), flavor_name=FLAVOR_NAME
+    )[ENV]
+    conda_yaml_path = os.path.join(model_dir, conda_yaml_filename)
     if format == "pip":
-        req_file_uri = append_to_uri_path(model_uri, _REQUIREMENTS_FILE_NAME)
-        try:
-            return _download_artifact_from_uri(req_file_uri)
-        except Exception as e:
-            # fallback to download conda.yaml file and parse the "pip" section from it.
-            _logger.info(
-                f"Downloading model '{_REQUIREMENTS_FILE_NAME}' file failed, error is {repr(e)}. "
-                "Falling back to fetching pip requirements from the model's 'conda.yaml' file. "
-                "Other conda dependencies will be ignored."
-            )
+        requirements_file = os.path.join(model_dir, _REQUIREMENTS_FILE_NAME)
```

Comment:
This line needs to be updated once #6684 is merged.

Reply:
https://github.com/mlflow/mlflow/pull/6684 is ready for merge, IMO :)

Reply:
@dbczumar Merged #6684. I'll update this PR, and then merge :)

Reply:
This line doesn't need to be updated, but line 494 does.